labshare/tests.py

```diff
@@ -463,6 +492,147 @@ def test_reservation_str_representation(self):
         ))
 
 
+class TestReservationExpiration(LabshareTestSetup):
+    @classmethod
+    def setUpTestData(cls):
+        LabshareTestSetup.setUpTestData()
+        cls.other_user = mommy.make(User, email="otheruser@example.com")
+        cls.other_user.groups.add(cls.group)
+
+    def assertTimeAlmostEqual(self, a, b):
+        self.assertLess(abs(a - b), timedelta(seconds=1))
+
+    def make_and_check_new_reservation(self, user, device_num=0):
+        previous_num_reservations = Reservation.objects.count()
+
+        response = self.app.get(reverse("reserve"), user=user)
+        self.assertEqual(response.status_code, 200)
+
+        form = response.form
+        form["device"].force_value(self.devices[device_num].name)
+        form["gpu"].force_value(self.devices[device_num].gpus.first().uuid)
+        form["next-available-spot"] = "false"
+
+        response = form.submit()
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(Reservation.objects.count(), previous_num_reservations + 1)
+
+    def test_new_reservation(self):
+        reservation = mommy.make(Reservation, gpu=self.gpu, user=self.user)
+        self.assertIsNone(reservation.usage_started)
+        self.assertIsNone(reservation.usage_expires)
+        self.assertFalse(reservation.is_usage_expired())
+
+        reservation.start_usage()
+        self.assertTimeAlmostEqual(reservation.usage_started, utc_now())
+        self.assertTimeAlmostEqual(reservation.usage_expires, utc_now() + Reservation.usage_period())
+
+    def test_expiring_reservation(self):
+        back_to_the_future = Reservation.usage_period() - (Reservation.reminder_period() + timedelta(minutes=1))
+        reservation = make_reservation_in_the_past(self.user, self.gpu, back_to_the_future)
+
+        self.assertFalse(reservation.is_usage_expired())
+        self.assertFalse(reservation.needs_reminder())
+
+        with mock.patch("django.utils.timezone.now") as mock_now:
+            mock_now.return_value = utc_now() + timedelta(minutes=2)
+            self.assertFalse(reservation.is_usage_expired())
+            self.assertTrue(reservation.needs_reminder())
+
+        with mock.patch("django.utils.timezone.now") as mock_now:
```

Comment:
no need to start another `with` block, although it looks more structured like this...